### PR TITLE
Feature/exclude forms from search

### PR DESCRIPTION
**Pull request by njseeto**
Block search engine crawling and indexing
Add `robots.txt` file to prevent web crawling.
Add `X-Robots-Tag` header to prevent search engine indexing.

The `X-Robots-Tag` header has the same effect as the meta tag `<meta name="robots" content="noindex,nofollow">`
except the former applies to all content types, not just HTML files.
`X-Robot-Tag: none` is equivalent to `noindex, nofollow`, telling robots not to index, follow links or cache.

Story: https://trello.com/c/hjW5ZHPM/1425-exclude-all-forms-from-search-results

### CODE REVIEW

File: app/controllers/robots_txts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    if disallow_all_crawlers?
+      render template: :disallow_all, layout: false, content_type: 'text/plain'
+    else
+      render body: nil, status: :not_found
+    end
+  end
+
+  private
+
+  def disallow_all_crawlers?
```
Comment (brenetic) [inline app/controllers/robots_txts_controller.rb:14]:
```suggestion
  def deny_all_crawlers?
```

File: app/controllers/robots_txts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    ENV['DISALLOW_ALL_WEB_CRAWLERS'].present?
```
Comment (brenetic) [inline app/controllers/robots_txts_controller.rb:15]:
```suggestion
    ENV['DENY_ALL_WEB_CRAWLERS'].present?
```

are we going to set a default value for this in the editor which injects it into the runner container? if not then we should set a default value here.

**Comment by brenetic** [comment]
good stuff. are we also going to add `<meta name="robots" content="noindex,nofollow">` to the templates?

**Comment by brenetic** [comment]
nevermind! i should read the PR message first 😛

**Review by tomas-stefano — APPROVED** [review]
🎸  Awesome middleware idea ;)

**Pull request merged**